broadinstitute/leonardo-cron-jobs
# Unlink disk when deleting runtime and apps [IA-2524]

Previously when the cron-job ran to delete a runtime or app it wasn't unlinking the disks attached to those deleted resources resulting in errors for users trying to make new resources since the disks were seen as attached. 

(also cleaned up a unit test around stopAfterCreation which is a removed property)

Added unit tests and tested running the cron-job locally

Steps to reproduce to test:
1. Create a new GCE Cluster in dev
2. Delete the VM through google console
3. Run Zombie-Monitor (had to change the 1hr wait period)
4. Check the DB

Before with a cluster in `Running` and successfully deleting the VM in google
<img width="1146" alt="Screen Shot 2021-03-02 at 10 15 21 AM" src="https://user-images.githubusercontent.com/54322292/109670185-dbc24f00-7b40-11eb-9730-ef8dc57b7cc7.png">

After running the cron-job the PD is set to `NULL` for that runtime
<img width="882" alt="Screen Shot 2021-03-02 at 10 17 23 AM" src="https://user-images.githubusercontent.com/54322292/109670286-f5fc2d00-7b40-11eb-8d4b-319a288e3caf.png">

Tested the K8'sCluster version making a Galaxy app and deleting the cluster in google and checking the DB before and after running zombie-monitor
<img width="1620" alt="Screen Shot 2021-03-02 at 3 26 57 PM" src="https://user-images.githubusercontent.com/54322292/109711201-97e53f00-7b6c-11eb-8178-59c09419733f.png">
